sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/Aggregation.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.expressions;
+
+import org.apache.spark.annotation.Evolving;
+
+import java.io.Serializable;
+
+/**
+ * Aggregation in SQL statement.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public class Aggregation implements Serializable {
```

Comment:
why does it need to be `Serializable`?

Reply:
and can be `final class`?

Reply:
I got Serialization problem if didn't implement `Serializable`.
```
Caused by: java.io.NotSerializableException: org.apache.spark.sql.connector.expressions.Aggregation
Serialization stack:
	- object not serializable (class: org.apache.spark.sql.connector.expressions.Aggregation, value: org.apache.spark.sql.connector.expressions.Aggregation@1ac25dbb)
	- field (class: scala.Some, name: value, type: class java.lang.Object)
	- object (class scala.Some, Some(org.apache.spark.sql.connector.expressions.Aggregation@1ac25dbb))
	- field (class: org.apache.spark.sql.execution.RowDataSourceScanExec, name: aggregation, type: class scala.Option)
	- object (class org.apache.spark.sql.execution.RowDataSourceScanExec, Scan org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCScan$$anon$1@651cedfb [DEPT#0,max(SALARY)#19,min(BONUS)#20] PushedAggregates: [Max(SALARY), Min(BONUS)], PushedFilters: [IsNotNull(DEPT), GreaterThan(DEPT,0)], PushedGroupby: [DEPT], ReadSchema: struct<DEPT:int,max(SALARY):decimal(20,2),min(BONUS):double>
)
	- field (class: org.apache.spark.sql.execution.aggregate.HashAggregateExec, name: child, type: class org.apache.spark.sql.execution.SparkPlan)
	- object (class org.apache.spark.sql.execution.aggregate.HashAggregateExec, HashAggregate(keys=[DEPT#0], functions=[partial_max(max(SALARY)#19), partial_min(min(BONUS)#20)], output=[DEPT#0, max#23, min#24])
+- Scan org.apache.spark.sql.execution.datasources.v2.jdbc.JDBCScan$$anon$1@651cedfb [DEPT#0,max(SALARY)#19,min(BONUS)#20] PushedAggregates: [Max(SALARY), Min(BONUS)], PushedFilters: [IsNotNull(DEPT), GreaterThan(DEPT,0)], PushedGroupby: [DEPT], ReadSchema: struct<DEPT:int,max(SALARY):decimal(20,2),min(BONUS):double>
)
```